### DISCUSSION TITLE
An error when tracking across frame

### DISCUSSION THREAD

**Discussion by hamember** [talmolab/sleap#1069] [Help!] [answered]
Hi!
I study bumble bee behaviour in Chinese Academy of Science. Sleap is marvellous!

Now we are done with training, but fail to track individuals across frame with both our own video and your established model presented with bumble bees. We tried in many PC but cannot manage it. It is OK to run inference without non tracker method.

Could you help me with this.    
![2](https://user-images.githubusercontent.com/119824925/205602475-cc1ca91d-190a-44f9-bddd-9c1564b5689a.png)
![未命名图片](https://user-images.githubusercontent.com/119824925/205602516-8a2612c4-0772-4c18-847d-355ae66ec5db.png)

Thanks!

Regards!

Jin

**Comment by roomrys** [comment]
Hi Jin,

Can you try tracking using the greedy matching method instead of Hungarian. 

**Update**: After troubleshooting a different dataset that yielded the same error as above, we found that the error was due to having `PredictedInstance`s returned with no points present (all points occluded). These empty instances resulted in an entire row of `np.inf` in our cost matrix which lead to an unsolvable optimization problem. See the PR #1073.

Thanks,
Liezl

**Reply by hamember** [reply]
Hi Liezl,

Thanks for your kind and rapid reply!

It seems that my model has some error, there is many redudantly trained frames. So, I trained a new model. This time I used both Hungarian and greedy matching for tracking. Both of them work when comined with simple tracking methods. But neither worked out with a flow tracking method.  Please see the screen shot

Of course, I can share my datasets. Please check the email. 

Thanks!

Jin
![221](https://user-images.githubusercontent.com/119824925/206126369-6d329f3a-349d-4ee1-a523-0fbc348b0fad.png)
![1](https://user-images.githubusercontent.com/119824925/206126404-0255fb1b-f55c-402e-a52f-bd96ac2c7ed0.png)

**Reply by roomrys** [reply]
Hi @hamember,

The OOM with flow tracker is likely caused by a bug introduced in #870 but later solved in #1001. #870 automatically saved shifted instances when using the flow tracker. #1001 ensures that the saved instances are removed from memory after a certain size is reached.

I will test this on the dataset you sent an follow up here.

Thanks,
Liezl

**Comment by roomrys** [comment, marked as answer]
Hi @hamember,

The original bug for `ValueError: cost matrix infeasible` for Hungarian matching has been fixed with #1073. This fix will be available in the next release, or you can get this fix ASAP by [installing/updating the source code](https://sleap.ai/installation.html#conda-from-source).

Testing on the dataset you sent, the OOM issue for flow tracker is isolated to SLEAP v1.2.8 and has been resolved in SLEAP v1.2.9.

Thanks,
Liezl
